tests/unit/test-cmd/test.build.js

```diff
@@ -37,6 +63,103 @@ describe('build', () => {
     );
   });
 
+  it('gives the correct name to a localized extension', () => {
+    return withTempDir(
+      (tmpDir) =>
+        build({
+          sourceDir: fixturePath('minimal-localizable-web-ext'),
+          artifactsDir: tmpDir.path(),
+        })
+        .then((buildResult) => {
+          assert.match(buildResult.extensionPath,
+                       /name_of_the_extension-1\.0\.zip$/);
+          return buildResult.extensionPath;
+        })
+    );
+  });
+
+  it('gives the correct name to a localized' +
+    'extension with repeating pattern in manifest', () => {
+    return withTempDir(
+      (tmpDir) => {
+        const messageFileName = path.join(tmpDir.path(), 'messages.json');
+        //This is missing the 'message' key
+        fs.writeFileSync(messageFileName,
+          `{"extensionName": {
+              "message": "example extension",
+              "description": "example extension"
+            }
+          }`);
+        return getDefaultLocalizedName({
+          messageFile: messageFileName,
+          manifestData: basicLocalizedManifestWithRepeatingPattern,
+        })
+          .then((result) => {
+            assert.match(result, /example extensionexample extension/);
```

Comment:
this would be easier to read if it had a space (or something) in it: `example extension example extension`